tests/phpunit/elementor/core/page-assets/test-loader.php

```diff
@@ -0,0 +1,202 @@
+		] );
+
+		$assets_loader->enqueue_assets();
+
+		$this->assertContains( 'e-style-dynamic-enqueue', $wp_styles->queue, 'e-style-dynamic-enqueue was not properly enqueued.' );
+		$this->assertContains( 'e-script-dynamic-enqueue', $wp_scripts->queue, 'e-script-dynamic-enqueue was not properly enqueued.' );
+	}
+
+	public function test_register_assets() {
+		global $wp_styles;
+		global $wp_scripts;
+
+		// Creating a new instance to get only the initial assets that exist in the assets loader (without the testing assets that were added by the other testing methods).
+		$new_assets_loader_instance = new Assets_Loader();
+
+		$current_assets = $new_assets_loader_instance->get_assets();
+
+		// Deregister the current assets.
+		foreach ( $current_assets as $assets_type => $assets_type_data ) {
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				if ( 'scripts' === $assets_type ) {
+					wp_deregister_script( $asset_name );
+					$this->assertArrayNotHasKey( $asset_name, $wp_scripts->registered, 'script asset should not be registered.' );
+				} else {
+					wp_deregister_style( $asset_name );
+					$this->assertArrayNotHasKey( $asset_name, $wp_styles->registered, 'style asset should not be registered.' );
+				}
+			}
+		}
+
+		// Creating additional instance in order to trigger the private 'register_assets' method that initiated by the Assets Loader constructor.
+		new Assets_Loader();
+
+		// Making sure that the additional assets loader instance has registered all the assets that were de-registered above.
+		foreach ( $current_assets as $assets_type => $assets_type_data ) {
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				if ( 'scripts' === $assets_type ) {
+					$this->assertArrayHasKey( $asset_name, $wp_scripts->registered, 'script asset was not properly registered.' );
+				} else {
+					$this->assertArrayHasKey( $asset_name, $wp_styles->registered, 'style asset was not properly registered.' );
+				}
+			}
+		}
+	}
+
+	private function add_testing_assets( $style_asset_key, $script_asset_key ) {
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$assets = $assets_loader->get_assets();
+
+		$this->assertArrayNotHasKey( $style_asset_key, $assets['styles'], 'Style asset key already exist, set a unique key for each style asset to prevent data collision.' );
+		$this->assertArrayNotHasKey( $script_asset_key, $assets['scripts'], 'Script asset key already exist, set a unique key for script each asset to prevent data collision.' );
+
+		$asset_data = [
+			'src' => 'path',
+			'version' => '5.9.1',
+			'dependencies' => [],
+		];
+
+		$test_assets = [];
+
+		$test_assets['styles'][ $style_asset_key ] = $asset_data;
+
+		$test_assets['scripts'][ $script_asset_key ] = $asset_data;
+
+		$assets_loader->add_assets( $test_assets );
+	}
+
+	private function add_optimized_assets_loading_experiment() {
+		$experiments = $this->elementor()->experiments;
+
+		$experiments->add_feature( [
```

Comment:
Why this feature doesn't exists?

Reply:
Because there are other tests that have removed it.